public/app/config/components.js

```diff
@@ -137,10 +137,34 @@
                 );
 
                 ko.components.register(
-                    'search/header',
+                    'search/basic',
```

Comment:
The `search/header` component has actually become `search/quick` (see below) which is more descriptive of what it does rather than where it happens to be used.  This `search/basic` component is the basic search on the search pages, then `search/query-builder` is the advanced search.  Finally `search/results` is the results view (and has a corresponding `controls/list-controls` component).  This decomposition makes the code more modular and easier to understand.